### DISCUSSION TITLE
how can I change data state

### DISCUSSION THREAD

**Discussion by engamir1** [flet-dev/flet#1068] [Q&A]
how can I change state through app pages and make listeners to data changes? 
is that by
`page. update() `

**Comment by iron3oxide** [comment]
In flet itself, each control has its own state. To create a listener/handler, define a function/method that takes at least one argument (commonly named `e` because it represents the event that is being received). Each flet control should have a set of properties that hold functions that are automatically called when the data of that control changes or some other interaction happens. A common example might be the `on_change` property. You can set that property to your listener/handler when you create the control or afterwards.
